src/badge/src/components/Badge.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+import React from 'react';
+import styled from 'styled-components';
+
+import * as Colors from '../../../colors';
+import { getStyleVariation } from '../../../utils/StyleUtils';
+
+const {
+  GREENS,
+  NEUTRALS,
+  PURPLES,
+  REDS,
+  WHITE,
+} = Colors;
+
+const getBackgroundColor = getStyleVariation('mode', {
+  added: GREENS[0],
+  primary: PURPLES[2],
+  removed: REDS[0],
+  secondary: PURPLES[5],
+  subtle: WHITE,
+}, NEUTRALS[6]);
+
+const getFontColor = getStyleVariation('mode', {
+  added: GREENS[3],
+  primary: WHITE,
+  removed: REDS[3],
+  secondary: PURPLES[1],
+  subtle: PURPLES[2],
+}, NEUTRALS[0]);
+
+const BadgeWrapper = styled.span`
+  background-color: ${getBackgroundColor};
+  border-radius: 32px;
+  color: ${getFontColor};
+  font-size: 12px;
+  font-weight: normal;
+  margin: 0 5px;
+  padding: 3px 8px;
+`;
+
+type Props = {
+  className ? :string;
+  count :number;
+  max ? :number;
+  mode ? :string;
+};
+
+const Badge = ({
+  className,
+  count,
```

Comment:
No change required at the moment, just some thoughts.

You are requiring users to pass in a `count` prop, which is fine, but it limits to only numbers. Instead, I think it'd be more useful to use `children` and if the child happens to be a number, apply the `max` logic. For example:

```
// render strings
<Badge>health</Badge>

// truncate strings
<Badge>reallylongstringnamethatshouldbetruncated</Badge>
=> <Badge>reallylong...</Badge>

// numbers work exactly the same as strings
<Badge>13</Badge>

// but we can also apply a max to the number
<Badge>100</Badge>
=> <Badge>99+</Badge>
```

Further, this sets us up to support actual components in the future if we wanted to.

Reply:
I'm actually fine with using `count` as a prop. This better differentiates a `Badge` from a `Tag` outside of just their appearances. If you want to use `children` or show text, you should present that information with a `Tag` instead. Otherwise, there would be no reason to have both. In my mind, a `Badge`'s only purpose is to show a numerical count.

I would rather extend the `Tag` to have truncation of text, clickable actions, icons, etc. This will also help the designers solidify their design language if we enforce a technical difference between the two. PSA already has designs that uses 'Tag' elements to show status and texts. We should be careful to respect the distinction between these elements.

https://zpl.io/VxYroAW
https://zpl.io/aRYOK6v

Reply:
Actually, I was originally just passing `children` to the `Badge` rather than a `count` prop, but then I went through the exact same thought process that you, @solomontang, just described and I changed the component to use `count`. I agree that allowing `Badges` only to display a numerical count differentiates it from a `Tag` and therefore merits its existence. I had that thought about passing components to a `Badge`, but I felt like it would make more sense to pass components to a `Tag` rather than a `Badge`.

Also, I was basing my decisions on Atlaskit's `Badge`, which appears to be used specifically for numerical purposes: https://atlaskit.atlassian.com/packages/core/badge. Looks like this `Badge` doesn't use a `count` prop (or similar), but I still feel that the prop should be used to differentiate it from `Tag`.

Reply:
Their default is `∞`. 🤔

Reply:
Yaaa kinda weird. 

Ya this is true, I'll just take out the +/ modes for the `Badges` for now, since we don't really need them.